content/en/cd-as-a-service/concepts/deployment/how-hpa-works.md

```diff
@@ -0,0 +1,73 @@
+ 
+ Kubernetes implements Horizontal Auto Scaling (HPA) as a control loop that runs intermittently. `HorizontalPodAutoscaler` controls the scale of a Deployment and its `ReplicaSet`. 
+ 
+ HPA is not recommended for replication controllers. 
+ 
+ HPA does not manage rolling updates by manipulating the replication controller. The deployment object manages the size of underlying replica sets. 
+
+ The sync interval is set by the `kube-controller-manager`. The default interval is 30 seconds. Change the default interval using the `horizontal-pod-autoscaler-sync-period` flag.
+
+The default delay is 3 minutes after an upscale event to allow metrics to stabilize. Change the default upscale delay period using the `horizontal-pod-autoscaler-upscale-delay` flag.
+ 
+## How HPA works in CD-as-a-Service
+When HPA is configured for an Armory Continuous Deployment-as-a-Service (CD-as-a-Service) deployment the deployment is converted to a `ReplicaSet` and the HPA configuration file is automatically re-written to reference the generated `ReplicaSet` resource. Continuous Deployment-as-a-Service freezes scaling behavior during a deployment.
+
+Consider a deployment to upgrade from the v1 to v2 version of your application. In this scenario the v1 application is running with 10 replicas and is scaled by HPA (min: 5 pods, max: 15 pods).
+ - At the beginning of a deployment, the HPA is deleted. Your v1 application has 10 pods but no longer scales between 5 and 15 pods.
+ - CD as a Service deploys the v2 instance of your application with 10 pods (20 total pods between v1 and v2).
+ - At the end of the deployment, a new HPA resource is created to target the v2 application.
+ - If the deployment needs to be rolled back, CDaaS deletes the v2 application and re-creates the original HPA.
+
+### Supported versions
+Armory Continuous Deployment-as-a-Service supports the following HorizontalPodAutoscaler API versions:
+- [v1](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/horizontal-pod-autoscaler-v1/)
+- [v2beta 1 and 2](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/horizontal-pod-autoscaler-v2beta2/)
+-  [v2](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/horizontal-pod-autoscaler-v2/)
+  
+**Before deployment**
+
+- Upgrade a database schema
+- Custom approval process
+
+**Within your deployment strategy**
+
+- Check logs and system health
+- Run custom tests
+
+**After deployment**
+Once CPU utilization drops to 0, the HPA automatically scales the number of replicas down to 1.
+
+> Autoscaling the replicas may take a few minutes.
+
+Run integration tests in a staging environment.
+- Perform metric tests
+- Run security scanners
+
+**Check the current status of the `HorizontalPodAutoscaler`**
+```
+kubectl get hpa
+```
+### Roll back an HPA configured deployment
+During a rollback, Armory CDaaS relies on the `kubectl.kubernetes.io/last-applied-configuration` annotation to re-create the HPA. 
```

Comment:
```suggestion
During a rollback, CD-as-a-Service relies on the `kubectl.kubernetes.io/last-applied-configuration` annotation to recreate the `HPA`. 
```

Reply:
I updated this without seeing your comment - please re-review :)